odoo/addons/base/models/ir_property.py

```diff
@@ -172,7 +173,7 @@ def get(self, name, model, res_id=False):
         if not res_id:
             t, v = self._get_default_property(name, model)
             if not v or t != 'many2one':
-                return v
+                return v if t else None
             return self.env[v[0]].browse(v[1])
```

Comment:
That `if `statement is unreadable.  What about negating it?
```python
if not t:
    return None
if t == 'many2one' and v:
    return self.env[v[0]].browse(v[1])
return v
```

Reply:
Yep of course.